Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,8 +133,29 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
-            if (parameter == null || parameter.IsReadOnly)
-                return false;
+            // Workset parameters
+            // Filter for worksets
+            if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)
+            {
+                // Skip non-workshared documents because they can't contain worksets
+                if (document.IsWorkshared == false)
```

Comment:
I would simplify to `if(!document.IsWorkshared)`

Reply:
Simplified as suggested

Reply:
This is not needed here at all actually - checked inside the `Create` method

Reply:
Removed as suggested